docs/user_workloads/creating_vms.md

```diff
@@ -161,18 +95,81 @@ Please note that volumes of different kinds currently have the following fixed
 boot order regardless of the order their flags were specified on the
 command line:
 
-1. ContainerDisk
-2. DataSource
-3. Cloned PVC
-4. Directly used PVC
+1. Containerdisks
+2. Directly used PVCs
+3. DataSources
+4. Cloned PVCs
+5. Blank volumes
+6. Imported volumes (through the `--volume-import` flag)
 
 If multiple volumes of the same kind were specified their order is
 determined by the order in which their flags were specified.
 
-## Specifying cloud-init user data
+## Generating cloud-init user data
+
+To generate cloud-init user data with `virtctl create vm` the following
+flags can be used.
+
+!!! Note
+    Generating cloud-init user data is mutually exclusive with [specifying custom cloud-init user data](#specifying-custom-cloud-init-user-data), as explained below.
+
+### `--user` flag
 
-To pass cloud-init user data to virtctl it needs to be encoded into a base64
-string. Here is an example how to do it:
+This flag allows to specify the main user of the virtual machine that is
+created by cloud-init. It sets the `user` parameter in the generated
+cloud-init user data.
+
+### `--password-file` flag
+
+This flag allows to specify a file to read the password for the virtual
+machine's main user from. In the generated cloud-init user data, it sets the
+value of the `password` parameter to the read in value and the value of the
+`chpasswd` parameter to `{ expire: False }`.
+
+### `--ssh-key` flag
+
+This flag allows to specify one or more SSH authorized keys for the virtual
+machine's main user. It sets the `ssh_authorized_keys` parameter in the
+generated cloud-init user data.
+
+### `--ga-manage-ssh` flag
+
+When this flag is set a command enabling the `qemu-guest-agent` to manage SSH
+authorized keys is added to the generated cloud-init user data. The command
+is added to the `runcmd` parameter which is required on SELinux enabled
+distributions that would otherwise not allow the `qemu-guest-agent` to manage
+SSH authorized keys in the home directories of users.
+
+### Example
+
+```shell
+$ virtctl create vm --user myuser --ssh-key THISISMYKEY --ga-manage-ssh
+```
+
+This command will generate the following cloud-init user data:
+
+```yaml
+volumes:
+  - cloudInitNoCloud:
+      userData: |-
+        #cloud-config
+        user: myuser
+        ssh_authorized_keys:
+          - THISISMYKEY
+        runcmd:
+          - [ setsebool, -P, 'virt_qemu_ga_manage_ssh', 'on' ]
```

Comment:
This is slightly confusing as `--ga-manage-ssh` isn't useful when a user provides `--ssh-key` right?

I'd break this out into another example or just explicitly use it in the access credentials example below.

Reply:
Ah yeah, well... not sure what I was thinking. This should be using `--access-cred`.